components/TextField/index.jsx

```diff
@@ -109,11 +119,54 @@ export default class TextField extends React.Component {
 
   // Property initializer binds method to class instance
   onChange = (...args) => {
-    this.updateValue(this.props.sanitizeValue(args[0].target.value));
+    const target = args[0].target;
+    this.updateValue(this.props.sanitizeValue(target.value));
     if (this.props.onChange) this.props.onChange(...args);
+
+
+    if (this.props.highlightList) {
+      this.setState({
+        highlightedContent: this.getHighlightedContent(target.value)
+      });
+
+      // In order to show the bad words correctly in a textarea
+      // we need to disable scrolling and expand the textarea
+      // dynamically.
+      if (this.props.multiLine) this.setDynamicTextAreaHeight(target);
+    }
+
     this.hideError();
   };
 
+  setDynamicTextAreaHeight = target => {
+    const oldHeight = Number(this.state.textAreaHeight);
+    const currentHeight = target.scrollHeight;
+    // Fix so when you backspace it will reduce the correct height
+    const newHeight = oldHeight > currentHeight ? currentHeight - 20 : currentHeight;
```

Comment:
This doesn't take into account all possible changes, e.g.: when a user deletes 2 lines at once. Maybe for now we can leave it like this, but in the future, this should probably be calculated more accurately